src/main/kotlin/nl/info/zac/documentcreation/BpmnDocumentCreationService.kt

```diff
@@ -0,0 +1,133 @@
+            createSmartDocumentForAttendedFlow(documentCreationDataAttended).let {
+                smartDocumentsService.createDocumentAttended(
+                    data = this,
+                    smartDocument = it
+                )
+            }
+        }.onFailure {
+            LOG.log(Level.WARNING, "Failed to create document with attended flow", it)
+        }.getOrThrow()
+
+    /**
+     * Creates a SmartDocument object for the attended flow.
+     * In this flow the description of the zaaktype of the zaak in the provided document creation data is used
+     * as the SmartDocuments template group.
+     */
+    private fun createSmartDocumentForAttendedFlow(creationDataUnattended: BpmnDocumentCreationDataAttended) =
+        SmartDocument(
+            selection = Selection(
+                templateGroup = creationDataUnattended.templateGroupName,
+                template = creationDataUnattended.templateName,
+            ),
+            variables = Variables(
+                // SmartDocuments use file extensions (without the leading `.`) instead of media types
+                // as the output format
+                outputFormats = listOf(
+                    OutputFormat(MediaTypes.Application.MS_WORD_OPEN_XML.extensions.first().drop(1))
+                ),
+                redirectMethod = HttpMethod.POST,
+                redirectUrl = documentCreationCallbackUrl(
+                    zaakUuid = creationDataUnattended.zaak.uuid,
+                    taskId = creationDataUnattended.taskId,
+                    informatieobjecttypeUuid = creationDataUnattended.informatieobjecttypeUuid,
+                    templateGroupName = creationDataUnattended.templateGroupName,
+                    templateName = creationDataUnattended.templateName,
+                    title = creationDataUnattended.title,
+                    description = creationDataUnattended.description,
+                    creationDate = creationDataUnattended.creationDate,
+                    userName = creationDataUnattended.author ?: loggedInUserInstance.get().getFullName(),
+                ).toString()
+            )
+        )
+
+    fun getInformationObjecttypeUuid(zaak: Zaak, templateGroupId: String, templateId: String) =
+        smartDocumentsTemplatesService.getInformationObjectTypeUUID(
+            zaakafhandelParametersUUID = zaak.zaaktype.extractUuid(),
+            templateGroupId = templateGroupId,
+            templateId = templateId
+        )
+
+    fun documentCreationCallbackUrl(
+        zaakUuid: UUID,
+        taskId: String?,
+        informatieobjecttypeUuid: UUID,
+        templateGroupName: String,
+        templateName: String,
+        title: String,
+        description: String?,
+        creationDate: ZonedDateTime,
+        userName: String,
+    ) =
+        documentCreationService.createDocumentCreationUriBulder(title, description, creationDate, userName).apply {
+            queryParam("templateName", templateName)
```

Comment:
I would expect the logic that knows the names of query params (or that knows about query params in the first place) to be in the DocumentCreationService or even a client service instead of here. Strange that a high level service class like this needs to know about such client details. But let's leave that for now I propose.